### PR TITLE
Update the default state of each service

### PR DESCRIPTION
**Pull request by nsharma-git**
# Description

Update the default service state for greenfield environment when using Python27. This error is not reproducible with python3. 

Error message from customer who is trying to do a greenfield installation with python27
```
fatal: [s2-conf-zoo01.sig.mgt.intellicentre.net.au]: FAILED! => {"msg": "The field 'vars' has an invalid value, which includes an undefined variable. The error was: 'dict object' has no attribute u'confluent-zookeeper.service'\n\nThe error appears to have been in '/home/ansibleadmin/cp-ansible-6.2.1-post/zookeeper.yml': line 14, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n - name: Determine Installation Pattern - Parallel or Serial\n ^ here\n"}
```

The code in the PR unblocked them.

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Verified locally

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

**Review by domenicbove — APPROVED** [review]
lgtm, if tested!

**Comment by erikgb** [comment]
@nsharma-git What is the issue you are trying to solve here? Can you please add a description to the PR - or create an issue? I have tested the current code in my local environment, and that works even in "greenfield" environments. The proposed changes makes the code less readable IMO, so I would like to know **why** we should do this.

**Comment by nsharma-git** [comment]
> @nsharma-git What is the issue you are trying to solve here? Can you please add a description to the PR - or create an issue? I have tested the current code in my local environment, and that works even in "greenfield" environments. The proposed changes makes the code less readable IMO, so I would like to know **why** we should do this.

Hi @erikgb This change unblocks a few user who are using python2.7

File: kafka_broker.yml
Comment (domenicbove) [inline kafka_broker.yml:16]: Maybe to make it more readable we could do 
```
        install_pattern: "{{ 'parallel' if service.state != 'running' or kafka_broker_deployment_strategy == 'parallel' else 'serial' }}"
      vars:
        service: "{{ ansible_facts.services[kafka_broker_service_name + '.service'] | default({'state':'unknown'}) }}"
  
```

Comment (domenicbove) [reply kafka_broker.yml:16]: @erikgb wdyt? The issue is some users on python27 are complaining that the default here 
```
"{{ ansible_facts.services[kafka_broker_service_name + '.service'].state | default('unknown') }}"
```
is failing because `ansible_facts.services[kafka_broker_service_name + '.service']` returns nothing, so i guess we need a default on that element before doing the `.state`

Comment (erikgb) [reply kafka_broker.yml:16]: @domenicbove I do not have a Python 2.7 environment to test in, but I am hoping we can avoid the duplication. 😄 Does the following work?

`service_state: "{{ ansible_facts.services[kafka_broker_service_name + '.service'] | default({})['state'] | default('unknown')}}"`

Comment (erikgb) [reply kafka_broker.yml:16]: But your suggestion also works - for a single task. But it used in multiple tasks, I would try to stick with a "fact".

Comment (domenicbove) [reply kafka_broker.yml:16]: @erikgb oh ya yours better! @nsharma-git can you test? I dont have python2.7 either. 

@nsharma-git Can you please include the error message from the customer in the description?

Comment (nsharma-git) [reply kafka_broker.yml:16]: Thanks @erikgb and @domenicbove . 
I have updated the description to give more context. Since we discovered the sensitive information is getting printed only when we use `diff` flag, I'm checking with security expert, if this is a real concern. Meanwhile, I'll verify  on python27.

Comment (erikgb) [reply kafka_broker.yml:16]: You should inform your customers about: https://www.python.org/doc/sunset-python-2/. They are running a great risk still on python 2....

**Comment by JumaX** [comment]
Looks like jenkins failed with the following:

--> [36mScenario: 'rbac-scram-custom-rhel'[0m
20:12:38  --> [36mAction: 'syntax'[0m
20:12:39  [31m/usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.[0m
20:12:39  [31m  from cryptography.exceptions import InvalidSignature[0m
20:12:41  [31mERROR! conflicting action statements: set_fact, service_state[0m
20:12:41  [31m[0m
20:12:41  [31mThe error appears to be in '/home/jenkins/workspace/c_Contributors_cp-ansible_PR-821/ksql.yml': line 14, column 7, but may[0m
20:12:41  [31mbe elsewhere in the file depending on the exact syntax problem.[0m
20:12:41  [31m[0m
20:12:41  [31mThe offending line appears to be:[0m
20:12:41  [31m[0m
20:12:41  [31m[0m
20:12:41  [31m    - name: Determine Installation Pattern - Parallel or Serial[0m
20:12:41  [31m      ^ here[0m
20:12:41  [31mERROR: [0m
20:12:41  [33mAn error occurred during the test sequence action: 'syntax'. Cleaning up.[0m
20:12:41  --> [36mScenario: 'rbac-scram-custom-rhel'[0m
20:12:41  --> [36mAction: 'cleanup'[0m
20:12:41  [33mSkipping, cleanup playbook not configured.[0m
20:12:41  --> [36mScenario: 'rbac-scram-custom-rhel'[0m
20:12:41  --> [36mAction: 'destroy'[0m
20:12:41  [31m/usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.[0m
20:12:41  [31m  from cryptography.exceptions import InvalidSignature[0m

**Comment by nsharma-git** [comment]
We are updating the doc for Python3. Will no longer support Python2

**Pull request closed without merging**
